test/e2e/e2e_test.go

```diff
@@ -131,6 +132,70 @@ var _ = ginkgo.Describe("JobSet", func() {
 		})
 	})
 
+	// This test is added to test the JobSet transitions as Kueue would when:
+	// doing: resume in RF1 -> suspend -> resume in RF2.
+	// In particular, Kueue updates the PodTemplate on suspending and resuming
+	// the JobSet.
+	ginkgo.When("JobSet is suspended and resumed", func() {
+
+		ginkgo.It("should allow to resume JobSet after updating PodTemplate", func() {
+			ctx := context.Background()
+			js := sleepTestJobSet(ns, 1).Obj()
+			jsKey := types.NamespacedName{Name: js.Name, Namespace: js.Namespace}
+
+			ginkgo.By("Create a suspended JobSet", func() {
+				js.Spec.Suspend = ptr.To(true)
+				js.Spec.TTLSecondsAfterFinished = ptr.To[int32](5)
+				gomega.Expect(k8sClient.Create(ctx, js)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Unsuspend the JobSet setting nodeSelectors that prevent pods from being scheduled", func() {
+				gomega.Eventually(func() error {
+					gomega.Expect(k8sClient.Get(ctx, jsKey, js)).Should(gomega.Succeed())
+					js.Spec.Suspend = ptr.To(false)
+					podTemplate := &js.Spec.ReplicatedJobs[0].Template.Spec.Template
+					if podTemplate.Spec.NodeSelector == nil {
+						podTemplate.Spec.NodeSelector = make(map[string]string)
+					}
+					podTemplate.Spec.NodeSelector["kubernetes.io/os"] = "non-existing-os"
+					return k8sClient.Update(ctx, js)
+				}, timeout, interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Await for all Jobs to be active", func() {
+				gomega.Eventually(func() int32 {
+					gomega.Expect(k8sClient.Get(ctx, jsKey, js)).Should(gomega.Succeed())
+					if js.Status.ReplicatedJobsStatus == nil {
+						return 0
+					}
+					return js.Status.ReplicatedJobsStatus[0].Active
```

Comment:
I think we should add a comment here explaining a Job is considered "active" by the JobSet controller if the `job.Status.Active > 0`, meaning if the Job's pending + running pods > 0. 

Since for other developers it is probably confusing and unintuitive that in the previous step we deliberately made the pods unschedulable yet now we expect the jobs to be "active." I know personally I've always found the Job API counting a Job's pending pods as "active" a bit odd, I feel like the name could be better, although naming things can be deceptively tricky :)

Reply:
Makes sense, added a comment to clarify that.